main/lsp/requests/hover.cc

```diff
@@ -24,11 +24,38 @@ string methodSignatureString(const core::GlobalState &gs, const core::TypePtr &r
     return contents;
 }
 
-unique_ptr<MarkupContent> formatRubyCode(MarkupKind markupKind, string str) {
-    if (markupKind == MarkupKind::Markdown && str.length() > 0) {
-        str = fmt::format("```ruby\n{}\n```", str);
+unique_ptr<MarkupContent> formatRubyCode(MarkupKind markupKind, string_view sigString,
+                                         optional<string_view> optionalDocString) {
+    string_view docString = optionalDocString.value_or("");
+    // Get rid of at most 1 trailing newline
+    if (docString.length() > 0 && docString.back() == '\n') {
+        docString.remove_suffix(1);
     }
-    return make_unique<MarkupContent>(markupKind, move(str));
+    if (sigString.length() > 0 && sigString.back() == '\n') {
+        sigString.remove_suffix(1);
+    }
+
+    string content = "";
+
+    // Add docs
+    absl::StrAppend(&content, docString);
+    if (docString.length() > 0)
+        absl::StrAppend(&content, "\n");
+
+    // Add sig
+    string formattedSigString = "";
+    if (markupKind == MarkupKind::Markdown && sigString.length() > 0) {
```

Comment:
Is `sigString` ever 0? Does it need to be `optional`?

Reply:
We could make `methodSignatureString` return `optional` -- makes more sense. Could `&dispatchResult` ever be null? That's when it seems to possibly return an empty string

Reply:
I don't think `dispatchResult` is ever `null`.

Reply:
You don't need to check if `sigString` is non-zero length. We should always have something; otherwise, it's a bug. You should remove this check.